pkg/server/singleprocess/logstream.go

```diff
@@ -17,29 +18,72 @@ import (
 // outside of that context.
 type singleProcessLogStreamProvider struct{}
 
-func (s *singleProcessLogStreamProvider) StartWriter(ctx context.Context, log hclog.Logger, state serverstate.Interface, job *serverstate.Job) (logstream.Tracker, error) {
-	return &singleProcessLogStreamTracker{
+func (s *singleProcessLogStreamProvider) StartWriter(ctx context.Context, log hclog.Logger, state serverstate.Interface, job *serverstate.Job) (logstream.Writer, error) {
+	return &singleProcessLogStreamWriter{
 		log: log,
 		job: job,
 	}, nil
 }
 
-// singleProcessLogStreamTracker implements logstream.Tracker
-type singleProcessLogStreamTracker struct {
+func (s *singleProcessLogStreamProvider) StartReader(ctx context.Context, log hclog.Logger, job *serverstate.Job) (logstream.Reader, error) {
+	var outputR *logbuffer.Reader
+
+	// NOTE(izaak): Not having an output buffer on the job isn't an error condition.
+	// Not sure when it would happen though.
```

Comment:
I'm preserving this behavior because I don't fully understand it, but if anyone does feel free to chime in.